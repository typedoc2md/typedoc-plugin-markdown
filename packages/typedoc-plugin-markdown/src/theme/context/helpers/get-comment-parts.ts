import { backTicks, link } from '@plugin/libs/markdown/index.js';
import { escapeChars } from '@plugin/libs/utils/index.js';
import { MarkdownThemeContext } from '@plugin/theme/index.js';
import { CommentDisplayPart, Reflection } from 'typedoc';

export function getCommentParts(
  this: MarkdownThemeContext,
  model: CommentDisplayPart[],
): string {
  const md: string[] = [];
  for (const part of model) {
    switch (part.kind) {
      case 'text':
        md.push(part.text);
        break;
      case 'code':
        md.push(part.text);
        break;
      case 'inline-tag':
        switch (part.tag) {
          case '@label':
          case '@inheritdoc':
            break;
          case '@link':
          case '@linkcode':
          case '@linkplain': {
            if (part.target) {
              let url: string | undefined;
              if (typeof part.target === 'string') {
                url = part.target;
              } else if ('id' in part.target) {
                if (this.router.hasUrl(part.target)) {
                  url = getReflectionUrl(this, part.target);
                }
                if (typeof url === 'undefined') {
                  let target = part.target.parent!;
                  while (!this.router.hasUrl(target)) {
                    target = target.parent!;
                  }
                  url = getReflectionUrl(this, target);
                }
              }
              const text =
                part.tag === '@linkcode'
                  ? backTicks(part.text)
                  : escapeChars(part.text);
              md.push(url ? link(text, url) : text);
            } else {
              md.push(escapeChars(part.text));
            }
            break;
          }
          default:
            md.push(escapeChars(part.text));
            break;
        }
        break;
      case 'relative-link':
        switch (typeof part.target) {
          case 'number': {
            const reflection = this.page.project.files.resolve(
              part.target,
              this.page.model.project,
            );
            let url: string | undefined;
            if (
              typeof reflection === 'object' &&
              this.router.hasUrl(reflection)
            ) {
              url = this.urlTo(reflection);
            } else {
              const fileName = this.page.project.files.getName(part.target);
              if (fileName) {
                const anchor = part.targetAnchor ? `#${part.targetAnchor}` : '';
                url = this.relativeURL(`_media/${fileName}${anchor}`);
              }
            }
            if (url) {
              md.push(url);
              break;
            }
          }
          // fall through
          case 'undefined':
            md.push(part.text);
            break;
        }
        break;
      default:
        md.push('');
    }
  }
  return md.join('');
}

export function getReflectionUrl(
  context: MarkdownThemeContext,
  reflection: Reflection,
) {
  return context.urlTo(reflection);
}
