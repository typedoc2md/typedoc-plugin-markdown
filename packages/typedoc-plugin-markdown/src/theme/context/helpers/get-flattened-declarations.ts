import { DeclarationReflection, ReflectionKind } from 'typedoc';

export function getFlattenedDeclarations(
  model: DeclarationReflection[],
  options?: { includeSignatures: boolean },
): DeclarationReflection[] {
  const getFlattenedDeclarations = (current: DeclarationReflection) => {
    return (current.type as any)?.declaration?.children?.reduce(
      (acc: DeclarationReflection[], child: DeclarationReflection) => {
        (child as any).originalName = child.name;
        const childObj = {
          ...child,
          name: `${current.name}.${child.name}`,
          getFullName: () => `${current.getFullName()}`,
          getFriendlyFullName: () => `${current.getFriendlyFullName()}`,
        } as DeclarationReflection;
        return parseDeclarations(childObj, acc);
      },
      [],
    );
  };

  const parseDeclarations = (
    current: DeclarationReflection,
    acc: DeclarationReflection[],
  ) => {
    const shouldFlatten = (current.type as any)?.declaration?.children;

    const isAccessor = current.kind === ReflectionKind.Accessor;

    if (options?.includeSignatures) {
      if (isAccessor) {
        const accessors: any[] = [];
        if (current.getSignature) {
          accessors.push({
            ...current,
            getFullName: () => `${current.getFullName()}`,
            getFriendlyFullName: () => `${current.getFriendlyFullName()}`,
            name: `get ${current.name}`,
            type: current.getSignature.type,
            comment: current.getSignature?.comment,
          });
        }
        if (current.setSignature) {
          accessors.push({
            ...current,
            getFullName: () => `${current.getFullName()}`,
            getFriendlyFullName: () => `${current.getFriendlyFullName()}`,
            name: `set ${current.name}`,
            type: current.setSignature.type,
            comment: current.setSignature?.comment,
          });
        }
        return [...acc, ...accessors];
      }

      if (current.signatures?.length) {
        const signatures = current.signatures.map((signature) => {
          return {
            ...current,
            getFullName: () => `${current.getFullName()}`,
            getFriendlyFullName: () => `${current.getFriendlyFullName()}`,
            name: signature.name,
            type: signature.type,
            comment: signature.comment,
          };
        });
        return [...acc, ...signatures];
      }
    }

    return shouldFlatten
      ? [...acc, current, ...getFlattenedDeclarations(current)]
      : [...acc, current];
  };

  return model.reduce(
    (acc: DeclarationReflection[], current: DeclarationReflection) =>
      parseDeclarations(current, acc),
    [],
  );
}
