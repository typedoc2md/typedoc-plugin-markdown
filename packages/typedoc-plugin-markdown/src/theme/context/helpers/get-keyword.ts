import { ReflectionKind } from 'typedoc';

export function getKeyword(model: ReflectionKind): string {
  const KEYWORD_MAP = {
    [ReflectionKind.Class]: 'class',
    [ReflectionKind.Enum]: 'enum',
    [ReflectionKind.Function]: 'function',
    [ReflectionKind.TypeAlias]: 'type',
  };
  return KEYWORD_MAP[model];
}
