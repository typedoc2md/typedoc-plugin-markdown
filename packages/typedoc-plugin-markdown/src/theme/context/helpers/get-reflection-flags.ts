import { backTicks } from '@plugin/libs/markdown/index.js';
import { MarkdownThemeContext } from '@plugin/theme/index.js';
import { ReflectionFlags } from 'typedoc';

export function getReflectionFlags(
  this: MarkdownThemeContext,
  reflectionFlags: ReflectionFlags,
): string {
  const result: string[] = [];
  if (reflectionFlags?.isAbstract) {
    result.push(backTicks('abstract'));
  }
  if (reflectionFlags?.isConst) {
    result.push(backTicks('const'));
  }
  if (reflectionFlags?.isPrivate) {
    result.push(backTicks('private'));
  }
  if (reflectionFlags?.isProtected) {
    result.push(backTicks('protected'));
  }
  if (reflectionFlags?.isReadonly) {
    result.push(backTicks('readonly'));
  }
  if (reflectionFlags?.isStatic) {
    result.push(backTicks('static'));
  }
  if (reflectionFlags?.isOptional) {
    result.push(backTicks('optional'));
  }
  return result.join(' ');
}
