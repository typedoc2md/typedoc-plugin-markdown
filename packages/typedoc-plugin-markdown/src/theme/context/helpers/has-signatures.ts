import { DeclarationReflection, ReflectionType } from 'typedoc';

export function hasSignatures(model: DeclarationReflection): boolean {
  return (
    !(model.type as ReflectionType)?.declaration?.children?.length &&
    (Boolean(model.signatures?.length) ||
      Boolean((model.type as ReflectionType)?.declaration?.signatures?.length))
  );
}
