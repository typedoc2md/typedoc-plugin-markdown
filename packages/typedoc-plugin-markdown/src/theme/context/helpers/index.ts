export * from './get-angle-bracket.js';
export * from './get-comment-parts.js';
export * from './get-declaration-type.js';
export * from './get-description-for-comment.js';
export * from './get-flattened-declarations.js';
export * from './get-hierarchy-type.js';
export * from './get-keyword.js';
export * from './get-modifier.js';
export * from './get-parameter-default-value.js';
export * from './get-project-name.js';
export * from './get-property-default-value.js';
export * from './get-reflection-flags.js';
export * from './get-return-type.js';
export * from './has-useful-type-details.js';
export * from './is-group-kind.js';
export * from './use-table-format.js';
