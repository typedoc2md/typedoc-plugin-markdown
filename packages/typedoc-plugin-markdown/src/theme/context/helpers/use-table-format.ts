import { MarkdownThemeContext } from '@plugin/theme/index.js';
import { ReflectionKind } from 'typedoc';

export function useTableFormat(
  this: MarkdownThemeContext,
  key:
    | 'properties'
    | 'parameters'
    | 'enums'
    | 'typeDeclarations'
    | 'propertyMembers',
  reflectionKind?: ReflectionKind,
): boolean {
  if (key === 'parameters') {
    return isTable(this.options.getValue('parametersFormat'));
  }
  if (key === 'properties') {
    if (isTable(this.options.getValue('propertiesFormat'))) {
      return true;
    }
    if (reflectionKind === ReflectionKind.TypeAlias) {
      return isTable(this.options.getValue('typeAliasPropertiesFormat'));
    }
    if (reflectionKind === ReflectionKind.Class) {
      return isTable(this.options.getValue('classPropertiesFormat'));
    }
    if (reflectionKind === ReflectionKind.Interface) {
      return isTable(this.options.getValue('interfacePropertiesFormat'));
    }
    return false;
  }
  if (key === 'enums') {
    return isTable(this.options.getValue('enumMembersFormat'));
  }
  if (key === 'propertyMembers') {
    return isTable(this.options.getValue('propertyMembersFormat'));
  }
  if (key === 'typeDeclarations') {
    return isTable(this.options.getValue('typeDeclarationFormat'));
  }
  return false;
}

function isTable(key: string) {
  return key.toLowerCase().includes('table');
}
