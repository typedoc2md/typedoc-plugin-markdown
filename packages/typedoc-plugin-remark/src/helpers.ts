import * as path from 'path';
import { PageKind, Reflection, ReflectionKind } from 'typedoc';
import { MarkdownPageEvent } from 'typedoc-plugin-markdown';
import { KIND_TO_STRING } from './options/constants.js';
import { RemarkPlugin } from './types/options.js';

export function getPlugins(page: MarkdownPageEvent, plugins: RemarkPlugin[]) {
  return plugins.reduce(
    (acc: [string, Record<string, any>][], plugin: RemarkPlugin) => {
      if (
        typeof plugin === 'object' &&
        'applyTo' in plugin &&
        'plugins' in plugin
      ) {
        if (plugin.applyTo.length === 1 && plugin.applyTo[0] === '*') {
          acc = [...acc, ...plugin.plugins];
        }

        if (
          plugin.applyTo.includes('Readme') &&
          page.pageKind === PageKind.Index
        ) {
          acc = [...acc, ...plugin.plugins];
        }
        if (
          plugin.applyTo.includes('Index') &&
          page.pageKind !== PageKind.Index &&
          (page.model as Reflection).kind === ReflectionKind.Project
        ) {
          acc = [...acc, ...plugin.plugins];
        }
        if (
          plugin.applyTo.includes(
            KIND_TO_STRING.get((page.model as Reflection).kind) as string,
          )
        ) {
          acc = [...acc, ...plugin.plugins];
        }
      } else {
        acc.push(plugin);
      }
      return acc;
    },
    [],
  );
}

export function getFullPath(name: string) {
  const isLocalPath = /^\.{1,2}\/|^\//.test(name as string);
  return isLocalPath ? path.resolve(process.cwd(), name as string) : name;
}
