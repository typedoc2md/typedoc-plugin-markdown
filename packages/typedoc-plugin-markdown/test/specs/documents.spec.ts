import { expectFileToEqual } from '../helpers.js';

describe(`typedoc-plugin-markdown (Integration /Documents)`, () => {
  describe(`Project`, () => {
    it(`should compile index page for project`, () => {
      expectFileToEqual('documents', ['modules', 'members'], 'README.md');
    });

    it(`should compile a project document`, () => {
      expectFileToEqual(
        'documents',
        ['modules', 'members'],
        'documents/Project-Document-1.md',
        1,
      );
    });
  });

  describe(`Module`, () => {
    it(`should compile index page for module`, () => {
      expectFileToEqual(
        'documents',
        ['modules', 'members'],
        'ModuleWithDocuments1/README.md',
        1,
      );
    });

    it(`should compile module documents`, () => {
      expectFileToEqual(
        'documents',
        ['modules', 'members'],
        'ModuleWithDocuments1/documents/MODULE_DOC.md',
        1,
      );
    });
  });

  describe(`Enum`, () => {
    it(`should compile index page for enum`, () => {
      expectFileToEqual(
        'documents',
        ['members'],
        'ModuleWithDocuments1/enumerations/EnumWithDocuments.md',
        1,
      );
    });

    it(`should compile a module document`, () => {
      expectFileToEqual(
        'documents',
        ['members'],
        'ModuleWithDocuments1/enumerations/documents/ENUM_DOC.md',
        1,
      );
    });
  });

  describe(`Class`, () => {
    it(`should compile index page for class`, () => {
      expectFileToEqual(
        'documents',
        ['members'],
        'ModuleWithDocuments1/classes/ClassWithDocuments.md',
        1,
      );
    });

    it(`should compile a module document`, () => {
      expectFileToEqual(
        'documents',
        ['members'],
        'ModuleWithDocuments1/classes/documents/CLASS_DOC.md',
        1,
      );
    });
  });

  describe(`Interface`, () => {
    it(`should compile index page for interface`, () => {
      expectFileToEqual(
        'documents',
        ['members'],
        'ModuleWithDocuments1/interfaces/InterfaceWithDocuments.md',
        1,
      );
    });

    it(`should compile a interface document`, () => {
      expectFileToEqual(
        'documents',
        ['members'],
        'ModuleWithDocuments1/interfaces/documents/INTERFACE_DOC.md',
        1,
      );
    });
  });

  describe(`Interface`, () => {
    it(`should compile index page for interface`, () => {
      expectFileToEqual(
        'documents',
        ['members'],
        'ModuleWithDocuments1/interfaces/InterfaceWithDocuments.md',
        1,
      );
    });

    it(`should compile a interface document`, () => {
      expectFileToEqual(
        'documents',
        ['members'],
        'ModuleWithDocuments1/interfaces/documents/INTERFACE_DOC.md',
        1,
      );
    });
  });
});
