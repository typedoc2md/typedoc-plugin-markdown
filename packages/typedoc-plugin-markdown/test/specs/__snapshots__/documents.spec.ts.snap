// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documents Class should compile a module document: (Output File Strategy "members") (Option Group "1") 1`] = `
"[typedoc-stubs](../../../README.md) / [ModuleWithDocuments1](../../README.md) / [ClassWithDocuments](../ClassWithDocuments.md) / CLASS\\_DOC

> Class Documentation

Document content
"
`;

exports[`Documents Class should compile index page for class: (Output File Strategy "members") (Option Group "1") 1`] = `
"[typedoc-stubs](../../README.md) / [ModuleWithDocuments1](../README.md) / ClassWithDocuments

# Class: ClassWithDocuments

## Documents

- [CLASS\\_DOC](documents/CLASS_DOC.md)

## Constructors

_CONSTRUCTOR_MEMBER_PARTIAL_
"
`;

exports[`Documents Enum should compile a module document: (Output File Strategy "members") (Option Group "1") 1`] = `
"[typedoc-stubs](../../../README.md) / [ModuleWithDocuments1](../../README.md) / [EnumWithDocuments](../EnumWithDocuments.md) / ENUM\\_DOC

> Enum Documentation

Document content
"
`;

exports[`Documents Enum should compile index page for enum: (Output File Strategy "members") (Option Group "1") 1`] = `
"[typedoc-stubs](../../README.md) / [ModuleWithDocuments1](../README.md) / EnumWithDocuments

# Enumeration: EnumWithDocuments

## Documents

- [ENUM\\_DOC](documents/ENUM_DOC.md)

## Enumeration Members

### EnumMember

_DECLARATION_TITLE_PARTIAL_
"
`;

exports[`Documents Interface should compile a interface document: (Output File Strategy "members") (Option Group "1") 1`] = `
"[typedoc-stubs](../../../README.md) / [ModuleWithDocuments1](../../README.md) / [InterfaceWithDocuments](../InterfaceWithDocuments.md) / INTERFACE\\_DOC

> Interface Documentation

Document content
"
`;

exports[`Documents Interface should compile a interface document: (Output File Strategy "members") (Option Group "1") 2`] = `
"[typedoc-stubs](../../../README.md) / [ModuleWithDocuments1](../../README.md) / [InterfaceWithDocuments](../InterfaceWithDocuments.md) / INTERFACE\\_DOC

> Interface Documentation

Document content
"
`;

exports[`Documents Interface should compile index page for interface: (Output File Strategy "members") (Option Group "1") 1`] = `
"[typedoc-stubs](../../README.md) / [ModuleWithDocuments1](../README.md) / InterfaceWithDocuments

# Interface: InterfaceWithDocuments

## Documents

- [INTERFACE\\_DOC](documents/INTERFACE_DOC.md)

## Properties

### prop

_DECLARATION_TITLE_PARTIAL_
"
`;

exports[`Documents Interface should compile index page for interface: (Output File Strategy "members") (Option Group "1") 2`] = `
"[typedoc-stubs](../../README.md) / [ModuleWithDocuments1](../README.md) / InterfaceWithDocuments

# Interface: InterfaceWithDocuments

## Documents

- [INTERFACE\\_DOC](documents/INTERFACE_DOC.md)

## Properties

### prop

_DECLARATION_TITLE_PARTIAL_
"
`;

exports[`Documents Module should compile index page for module: (Output File Strategy "members") (Option Group "1") 1`] = `
"[typedoc-stubs](../README.md) / ModuleWithDocuments1

# ModuleWithDocuments1

## Documents

- [MODULE\\_DOC](documents/MODULE_DOC.md)

## Namespaces

- [NamespaceWithDocuments](namespaces/NamespaceWithDocuments/README.md)

## Enumerations

- [EnumWithDocuments](enumerations/EnumWithDocuments.md)

## Classes

- [ClassWithDocuments](classes/ClassWithDocuments.md)

## Interfaces

- [InterfaceWithDocuments](interfaces/InterfaceWithDocuments.md)

## Type Aliases

- [TypeWithDocuments](type-aliases/TypeWithDocuments.md)

## Variables

- [variableWithDocuments](variables/variableWithDocuments.md)

## Functions

- [functionWithDocuments](functions/functionWithDocuments.md)
- [multipleFunctionWithDocuments](functions/multipleFunctionWithDocuments.md)
"
`;

exports[`Documents Module should compile index page for module: (Output File Strategy "modules") (Option Group "1") 1`] = `
"[typedoc-stubs](../README.md) / ModuleWithDocuments1

# ModuleWithDocuments1

## Documents

- [MODULE\\_DOC](documents/MODULE_DOC.md)

## Namespaces

- [NamespaceWithDocuments](namespaces/NamespaceWithDocuments/README.md)

## Enumerations

### EnumWithDocuments

#### Documents

- [ENUM\\_DOC](documents/Enumeration.ENUM_DOC.md)

#### Enumeration Members

##### EnumMember

_DECLARATION_TITLE_PARTIAL_

## Classes

### ClassWithDocuments

#### Documents

- [CLASS\\_DOC](documents/Class.CLASS_DOC.md)

#### Constructors

_CONSTRUCTOR_MEMBER_PARTIAL_

## Interfaces

### InterfaceWithDocuments

#### Documents

- [INTERFACE\\_DOC](documents/Interface.INTERFACE_DOC.md)

#### Properties

##### prop

_DECLARATION_TITLE_PARTIAL_

## Type Aliases

### TypeWithDocuments

_DECLARATION_TITLE_PARTIAL_

#### Documents

- [TYPE\\_DOC](documents/TypeAlias.TYPE_DOC.md)

## Variables

### variableWithDocuments

_DECLARATION_TITLE_PARTIAL_

#### Documents

- [VARIABLE\\_DOC](documents/Variable.VARIABLE_DOC.md)

## Functions

### functionWithDocuments()

_SIGNATURE_TITLE_PARTIAL_

Comments for functionWithDocuments

#### Documents

- [FUNCTION\\_DOC\\_1](documents/Function.FUNCTION_DOC_1.md)

_SIGNATURE_RETURNS_PARTIAL_

***

### multipleFunctionWithDocuments()

Comments for multipleFunctionWithDocuments

#### Documents

- [FUNCTION\\_DOC\\_3](documents/Function.FUNCTION_DOC_3.md)
- [FUNCTION\\_DOC\\_1](documents/Function.FUNCTION_DOC_1-1.md)
- [FUNCTION\\_DOC\\_2](documents/Function.FUNCTION_DOC_2.md)

#### Call Signature

_SIGNATURE_TITLE_PARTIAL_

Comments for multipleFunctionWithDocumentsA

##### Parameters

_PARAMETERS_LIST_PARTIAL_

_SIGNATURE_RETURNS_PARTIAL_

#### Call Signature

_SIGNATURE_TITLE_PARTIAL_

Comments for multipleFunctionWithDocumentsB

##### Parameters

_PARAMETERS_LIST_PARTIAL_

_SIGNATURE_RETURNS_PARTIAL_
"
`;

exports[`Documents Module should compile module documents: (Output File Strategy "members") (Option Group "1") 1`] = `
"[typedoc-stubs](../../README.md) / [ModuleWithDocuments1](../README.md) / MODULE\\_DOC

> Module Documentation

Document content
"
`;

exports[`Documents Module should compile module documents: (Output File Strategy "modules") (Option Group "1") 1`] = `
"[typedoc-stubs](../../README.md) / [ModuleWithDocuments1](../README.md) / MODULE\\_DOC

> Module Documentation

Document content
"
`;

exports[`Documents Project should compile a project document: (Output File Strategy "members") (Option Group "1") 1`] = `
"[typedoc-stubs](../README.md) / Project Document 1

## Project Document 1

Document content
"
`;

exports[`Documents Project should compile a project document: (Output File Strategy "modules") (Option Group "1") 1`] = `
"[typedoc-stubs](../README.md) / Project Document 1

## Project Document 1

Document content
"
`;

exports[`Documents Project should compile index page for project: (Output File Strategy "members") (Option Group "1") 1`] = `
"# typedoc-stubs

## Documents

- [Project Document 1](documents/Project-Document-1.md)
- [docs/project/PROJECT\\_DOC\\_2](documents/docs/project/PROJECT_DOC_2.md)
- [docs/project/PROJECT\\_DOC\\_3](documents/docs/project/PROJECT_DOC_3.md)

## Modules

- [ModuleWithDocuments1](ModuleWithDocuments1/README.md)
- [ModuleWithDocuments2](ModuleWithDocuments2/README.md)
"
`;

exports[`Documents Project should compile index page for project: (Output File Strategy "members") (Option Group "2") 1`] = `
"# typedoc-stubs

## Documents

<table>
<thead>
<tr>
<th>Document</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>

[Project Document 1](Document.Project-Document-1.md)

</td>
<td>

A description of the document

</td>
</tr>
<tr>
<td>

[docs/project/PROJECT\\_DOC\\_2](Document.docs.project.PROJECT_DOC_2.md)

</td>
<td>

&hyphen;

</td>
</tr>
<tr>
<td>

[docs/project/PROJECT\\_DOC\\_3](Document.docs.project.PROJECT_DOC_3.md)

</td>
<td>

&hyphen;

</td>
</tr>
</tbody>
</table>

## Modules

<table>
<thead>
<tr>
<th>Module</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>

[ModuleWithDocuments1](ModuleWithDocuments1.md)

</td>
<td>

&hyphen;

</td>
</tr>
<tr>
<td>

[ModuleWithDocuments2](ModuleWithDocuments2.md)

</td>
<td>

&hyphen;

</td>
</tr>
</tbody>
</table>
"
`;

exports[`Documents Project should compile index page for project: (Output File Strategy "modules") (Option Group "1") 1`] = `
"# typedoc-stubs

## Documents

- [Project Document 1](documents/Project-Document-1.md)
- [docs/project/PROJECT\\_DOC\\_2](documents/docs/project/PROJECT_DOC_2.md)
- [docs/project/PROJECT\\_DOC\\_3](documents/docs/project/PROJECT_DOC_3.md)

## Modules

- [ModuleWithDocuments1](ModuleWithDocuments1/README.md)
- [ModuleWithDocuments2](ModuleWithDocuments2/README.md)
"
`;

exports[`Documents Project should compile index page for project: (Output File Strategy "modules") (Option Group "2") 1`] = `
"# typedoc-stubs

## Documents

<table>
<thead>
<tr>
<th>Document</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>

[Project Document 1](Document.Project-Document-1.md)

</td>
<td>

A description of the document

</td>
</tr>
<tr>
<td>

[docs/project/PROJECT\\_DOC\\_2](Document.docs.project.PROJECT_DOC_2.md)

</td>
<td>

&hyphen;

</td>
</tr>
<tr>
<td>

[docs/project/PROJECT\\_DOC\\_3](Document.docs.project.PROJECT_DOC_3.md)

</td>
<td>

&hyphen;

</td>
</tr>
</tbody>
</table>

## Modules

<table>
<thead>
<tr>
<th>Module</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>

[ModuleWithDocuments1](ModuleWithDocuments1.md)

</td>
<td>

&hyphen;

</td>
</tr>
<tr>
<td>

[ModuleWithDocuments2](ModuleWithDocuments2.md)

</td>
<td>

&hyphen;

</td>
</tr>
</tbody>
</table>
"
`;
